netty-mqtt/src/test/java/org/thingsboard/mqtt/MqttClientTest.java

```diff
@@ -119,6 +119,36 @@ void testConnectToBroker() {
         assertThat(client.isConnected()).isTrue();
     }
 
+    @Test
+    void testDisconnectFromBroker() {
+        // GIVEN
+        var clientConfig = new MqttClientConfig();
+        clientConfig.setOwnerId("Test[ConnectToBroker]");
+        clientConfig.setClientId("connect");
+
+        client = MqttClient.create(clientConfig, null, handlerExecutor);
+
+        // WHEN
+        Promise<MqttConnectResult> connectFuture = client.connect(broker.getHost(), broker.getMqttPort());
+
+        // THEN
+        assertThat(connectFuture).isNotNull();
+
+        Awaitility.await("waiting for client to connect")
+                .atMost(Duration.ofSeconds(10L))
+                .until(connectFuture::isDone);
+
+        assertThat(connectFuture.isSuccess()).isTrue();
```

Comment:
```suggestion
        connect(broker.getHost(), broker.getMqttPort());
```
Explicitly writing all these checks is necessary only in the test that verifies connection logic. In all other tests, use a private utility method that simply connects to a broker, assuming it will succeed. Connection errors should be caught by the test that checks connection logic. Also, when using the given-when-then scheme, avoid multiple “when” steps whenever possible. It’s very rarely needed and often indicates you’re either testing too much or treating a precondition as a “when”